src/AzureClient/AzureClient.cs

```diff
@@ -19,13 +20,27 @@
 using Microsoft.Jupyter.Core;
 using Microsoft.Quantum.IQSharp.Common;
 using Microsoft.Quantum.IQSharp.Jupyter;
+using Microsoft.Quantum.Runtime;
+using Microsoft.Quantum.Runtime.Submitters;
 using Microsoft.Quantum.Simulation.Common;
 
 namespace Microsoft.Quantum.IQSharp.AzureClient
 {
     /// <inheritdoc/>
     public class AzureClient : IAzureClient
     {
+        private const string MicrosoftSimulator = "microsoft.simulator";
+
+        /// <summary>
+        /// Returns whether a target ID is meant for quantum execution since not all targets
+        /// exposed by providers are meant for that. More specifically, the Microsoft provider exposes
+        /// targets that are not meant for quantum execution and the only ones meant for that start
+        /// with "microsoft.simulator"
+        /// </summary>
+        private static bool IsQuantumExecutionTarget(string targetId) =>
+            AzureExecutionTarget.GetProvider(targetId) != AzureProvider.Microsoft
```

Comment:
The API doesn't seem to be accessible from IQ# at this time.

Reply:
Sounds good, thanks for checking on that. Looks like it may be exposed through https://github.com/Azure/azure-sdk-for-net/blob/0a6567e46231bdef2d4963c9033411564943a0ea/sdk/quantum/Microsoft.Azure.Management.Quantum/src/Generated/QuantumManagementClient.cs#L82, which uses the public interface `IOfferingsOperations` rather than the private implementation `OfferingsOperations`, but that will take some significant work to add. Would you be willing to open an issue for that work and link it in a comment here? Thanks!